pyo3-build-config/src/impl_.rs

```diff
@@ -677,32 +787,22 @@ pub fn cross_compiling(
     target_os: &str,
 ) -> Result<Option<CrossCompileConfig>> {
     let env_vars = cross_compile_env_vars();
+    let target_info = TargetInfo::from_triple(target_arch, target_vendor, target_os, None);
 
-    let target_info = TargetInfo {
```

Comment:
I think the intent of open-coding the construction of the struct was to explicitly avoid being able to mess up the positional arguments to a function like `TargetInfo::from_triple`.

Reply:
I've added an optional fourth field and it was getting really noisy syntax-wise.
`from_triple()` is a private method that is only ever called by unit test code.
It is also called from a public method that PyO3 itself does not seem to use...

Reply:
I think there's compromise either way; I'm fine with `from_triple` like this.

Reply:
I'll probably revert this because I'll be resubmitting with many updates anyways...